crates/sui-rpc/src/client/v2/coin_selection.rs

```diff
@@ -25,17 +26,18 @@ impl Client {
     /// Returns an error if there are insufficient funds to meet the required amount or if there is an RPC error
     pub async fn select_coins(
         &self,
-        owner_address: impl Into<String>,
-        coin_type: impl Into<String>,
-        amount: impl Into<u128>,
-        exclude: &[&str],
+        owner_address: impl Into<Address>,
+        coin_type: impl Into<TypeTag>,
+        amount: impl Into<u64>,
+        exclude: &[Address],
     ) -> Result<Vec<Object>> {
+        let owner_address = owner_address.into();
         let coin_type = coin_type.into();
-        let owner_address_str = owner_address.into();
         let amount = amount.into();
+        let coin_struct = StructTag::coin(coin_type);
         let list_request = ListOwnedObjectsRequest::default()
-            .with_owner(&owner_address_str)
-            .with_object_type(format!("0x2::coin::Coin<{coin_type}>"))
+            .with_owner(owner_address.to_string())
+            .with_object_type(coin_struct.to_string())
```

Comment:
I *think* you can just pass owner and coin_strust without calling to_string, these methods should be able to do the conversion for you.